### PR TITLE
schema_grant: refactor schema grant Update function

### PR DESCRIPTION
**Pull request by igungor**
Related to #469

### CODE REVIEW

File: pkg/resources/schema_grant.go [hunk 1]
```diff
@@ -61,14 +61,12 @@ var schemaGrantSchema = map[string]*schema.Schema{
...
 		Optional:    true,
 		Description: "Grants privilege to these shares (only valid if on_future is unset).",
-		ForceNew:    true,
 	},
 	"on_future": {
 		Type:          schema.TypeBool,
 		Optional:      true,
 		Description:   "When this is set to true, apply this grant on all future schemas in the given database. The schema_name and shares fields must be unset in order to use on_future.",
 		Default:       false,
-		ForceNew:      true,
```
Comment (edulop91) [inline pkg/resources/schema_grant.go:71]: Should this remain `true`?

Comment (igungor) [reply pkg/resources/schema_grant.go]: you're right. reverted.

**Review by edulop91 — APPROVED** [review]

**Pull request merged**